homeassistant/components/weheat/sensor.py

```diff
@@ -0,0 +1,95 @@
+"""Platform for sensor integration."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from decimal import Decimal
+
+from weheat.abstractions.heat_pump import HeatPump
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfPower
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DISPLAY_PRECISION_COP, DISPLAY_PRECISION_WATTS
+from .coordinator import WeheatDataUpdateCoordinator
+from .entity import WeheatEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class WeHeatSensorEntityDescription(SensorEntityDescription):
+    """Describes Weheat sensor entity."""
+
+    value_fn: Callable[[HeatPump], Decimal]
```

Comment:
```suggestion
    value_fn: Callable[[HeatPump], StateType]
```
Any reason why it wasn't typed as `float`